src/backend/access/table/tableam.c

```diff
@@ -288,7 +288,7 @@ table_index_fetch_tuple_check(Relation rel,
 	 * overhead is significant for AO/CO relations. For details, please refer to
 	 * table_index_unique_check().
 	 */
-	if (RelationIsAppendOptimized(rel))
+	if (rel->rd_tableam->index_unique_check)
```

Comment:
Update comments?

Reply:
Updated. The current comment is good enough about the changes. The code change applies for AO/CO and other non-heap tables(including pax) that per-tuple overhead is expensive.